### ISSUE TITLE
Closing "ansi-term" buffers when sole window results in an error

### ISSUE THREAD

**Issue by joelmccracken** [syl20bnr/spacemacs#5629]
#### Description

When closing ansi-term when it takes up the full window, and error is raised (and it makes the closing take a few seconds instead of being instantaneous).

#### Reproduction guide
- Start Emacs
- Have the shell layer loaded. On my system, this looks like:
```
     (shell :variables
            shell-default-height 30
            shell-default-position 'bottom)
```

*Observed behaviour:*
I frequently use ansi-term as part of my workflow. When I close a terminal, say with C-d, emacs freezes for a second or so and reports an error.

*Expected behaviour:*
I expect it to close immediately.

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.14
- Spacemacs branch: master (rev. a298a5a)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
(emacs-lisp git markdown org
            (shell :variables shell-default-height 30 shell-default-position 'bottom)
            version-control rust elm racket elixir osx haskell)
```

#### Backtrace
```
```

### Patch

I have a patch that fixes this problem locally. I'll make a PR for it immediately after this.

**Comment by joelmccracken** [comment]
PR created #5630

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
